server/graphql/v2/mutation/CreateCollectiveMutation.js

```diff
@@ -22,96 +24,125 @@ async function createCollective(_, args, req) {
 
   const { remoteUser, loaders } = req;
 
-  if (!remoteUser) {
-    throw new Unauthorized('You need to be logged in to create a collective');
-  }
-
-  const collectiveData = {
-    slug: args.collective.slug.toLowerCase(),
-    ...pick(args.collective, ['name', 'description', 'tags']),
-    isActive: false,
-    CreatedByUserId: remoteUser.id,
-    settings: { ...DEFAULT_COLLECTIVE_SETTINGS, ...args.collective.settings },
-  };
-
-  if (isCollectiveSlugReserved(collectiveData.slug)) {
-    throw new Error(`The slug '${collectiveData.slug}' is not allowed.`);
-  }
-  const collectiveWithSlug = await models.Collective.findOne({ where: { slug: collectiveData.slug } });
-  if (collectiveWithSlug) {
-    throw new Error(`The slug ${collectiveData.slug} is already taken. Please use another slug for your collective.`);
-  }
+  let user = remoteUser,
+    host;
 
-  let host;
-
-  // Handle GitHub automated approval and apply to the Open Source Collective Host
-  if (args.automateApprovalWithGithub && args.collective.githubHandle) {
-    const githubHandle = args.collective.githubHandle;
-    const opensourceHost = defaultHostCollective('opensource');
-    host = await loaders.Collective.byId.load(opensourceHost.CollectiveId);
-    try {
-      const githubAccount = await models.ConnectedAccount.findOne({
-        where: { CollectiveId: remoteUser.CollectiveId, service: 'github' },
-      });
-      if (!githubAccount) {
-        throw new Error('You must have a connected GitHub Account to create a collective with GitHub.');
-      }
-      // In e2e/CI environment, checkGithubAdmin and checkGithubStars will be stubbed
-      await github.checkGithubAdmin(githubHandle, githubAccount.token);
-      await github.checkGithubStars(githubHandle, githubAccount.token);
-      shouldAutomaticallyApprove = true;
-    } catch (error) {
-      throw new ValidationFailed(error.message);
-    }
-    if (githubHandle.includes('/')) {
-      collectiveData.settings.githubRepo = githubHandle;
-    } else {
-      collectiveData.settings.githubOrg = githubHandle;
-    }
-    collectiveData.tags = collectiveData.tags || [];
-    if (!collectiveData.tags.includes('open source')) {
-      collectiveData.tags.push('open source');
-    }
-  } else if (args.host) {
+  if (args.host) {
     host = await fetchAccountWithReference(args.host, { loaders });
-    if (!host) {
-      throw new ValidationFailed('Host Not Found');
-    }
-    if (!host.isHostAccount) {
-      throw new ValidationFailed('Host account is not activated as Host.');
-    }
   }
 
-  const collective = await models.Collective.create(collectiveData);
+  return sequelize
+    .transaction(async transaction => {
+      if (!user && args.user && host.id === defaultHostCollective('foundation').CollectiveId) {
```

Comment:
Otherwise it will crash if no host is specified

```suggestion
      if (!user && args.user && host?.id === defaultHostCollective('foundation').CollectiveId) {
```